nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -511,7 +538,16 @@ in {
     };
 
     systemd.services.mastodon-init-db = lib.mkIf cfg.automaticMigrations {
-      script = ''
+      script = lib.optionalString (!databaseActuallyCreateLocally) ''
+        umask 077
+
+        export PGPASSFILE
+        PGPASSFILE=$(mktemp)
+        cat > $PGPASSFILE <<EOF
+        ${cfg.database.host}:${toString cfg.database.port}:${cfg.database.name}:${cfg.database.user}:$(cat ${cfg.database.passwordFile})
+        EOF
+
+      '' + ''
```

Comment:
Why add two strings instead of writing a long string?

Reply:
You mean this code?
```
        cat > $PGPASSFILE <<EOF
        ${cfg.database.host}:${toString cfg.database.port}:${cfg.database.name}:${cfg.database.user}:$(cat ${cfg.database.passwordFile})
        EOF
```

Reply:
No I mean why line 550 exists when all it does is concatenate two constant strings.

Reply:
This is done in such a way that this code will only work when `databaseActuallyCreateLocally = false`:
```
        umask 077
...
```
And this code worked in both variants  `databaseActuallyCreateLocally`:
```
        if [ `psql ${cfg.database.name} -c \
...
```

Reply:
Ah yes thanks! My bad.